Robot-Framework/test-suites/bat-tests/gui-tests.robot

```
@@ -0,0 +1,189 @@
+    Execute Command   ydotool click 0xC0  sudo=True  sudo_password=${PASSWORD}
+
+    Check that the application is not running    ${app}   5
+
+    # In case closing the app via GUI failed
+    [Teardown]    Run Keywords    Kill process  @{app_pids}
+    ...           AND             Move cursor to corner
+    ...           AND             Stop ydotoold
+
+Locate image on screen
+    [Documentation]    Take a screenshot. Locate given image on the screenshot.
+    ...                Return center coordinates of the image in mouse coordinate system
+    [Arguments]        ${image_to_be_searched}  ${confidence}=0.999   ${iterations}=5
+    ${coordinates}=  Set Variable  ${EMPTY}
+    ${pass_status}=  Set Variable  FAIL
+    FOR   ${i}   IN RANGE  ${iterations}
+        Log To Console    Taking screenshot
+        Execute Command   grim screenshot.png
+        SSHLibrary.Get File  screenshot.png   screenshot.png
+        Log To Console    Locating image on screenshot
+        ${pass_status}  ${coordinates}  Run Keyword And Ignore Error  Locate image  ${image_to_be_searched}   ${confidence}
+        IF    $pass_status=='PASS'    BREAK
+    END
+    IF    $pass_status=='FAIL'    FAIL  Image recognition failure: ${image_to_be_searched}
+    Log To Console    Coordinates: ${coordinates}
+    ${mouse_x}  Get From Dictionary   ${coordinates}  x
+    ${mouse_y}  Get From Dictionary   ${coordinates}  y
+    RETURN  ${mouse_x}  ${mouse_y}
+
+Start ydotoold
+    [Documentation]    Start ydotool daemon if it is not already running.
+    ${ydotoold_state}=    Execute Command    sh -c 'ps aux | grep ydotoold | grep -v grep'
+    IF  $ydotoold_state == '${EMPTY}'
+        Log To Console    Starting ydotool daemon
+        Run Keyword And Ignore Error  Execute Command   -b /run/current-system/sw/bin/ydotoold --socket-path /tmp/.ydotool_socket  sudo=True  sudo_password=${PASSWORD}  timeout=3
```

why do we ignore error? what kind of error can occur which we want to ignore?

---

This was a bit tricky part. The thing is that without forcing the timeout (and consequently failing the 'Execute Command' keyword) I could not make the execution to continue. Without timeout robot will just get stuck. When tried manually in gui-vm it requires one hit of enter to return to command prompt. I admit that this is not the best solution but I could not find better for now. Any suggestions?

In the current state this keyword cannot fail even if starting ydotool daemon failed for some reason. If starting the daemon didn't succeed it would show up at next `Execute Command ydotool mousemove...` which would fail. 

It's possible add 
`${ydotoold_state}=    Execute Command    sh -c 'ps aux | grep ydotoold | grep -v grep'` 
`Should Not Be Empty  ${ydotoold_state}  failed to start ydotool daemon`
also to the end of the keyword so that it would actually be able to fail if launching ydotoold didin't succeed.

---

I added the check that launching ydotool daemon succeeded.